salt/utils/subprocess.py

```diff
@@ -4,14 +4,50 @@
 '''
 from __future__ import absolute_import, print_function, unicode_literals
 
+import os
+import sys
 import shlex
+import logging
 import subprocess
 import threading
 import salt.exceptions
 import salt.utils.data
 from salt.ext import six
 
 
+log = logging.getLogger(__name__)
+
+
+class FdPopen(subprocess.Popen):
+    '''
+    Python2 closing file descriptors in a blind way by a system's given range.
+    Therefore close_fds can significantly slow down the system in various circumstances.
+    '''
+
+    if sys.platform.startswith('linux') and six.PY2:
+        def _close_fds(self, but):
+            '''
+            Close only file descriptors that are needed.
+            The issue is not new: https://bugs.python.org/issue1663329
+
+            :param but:
+            :return:
+            '''
+            maxfds = len(os.listdir('/proc/{}/fd'.format(os.getpid())))
```

Comment:
We should have a fallback to the existing behavior for corner cases where /proc is not mounted.